### ISSUE TITLE
[Web][a11y] Screen reader is reading widget spans with role 'link' as role 'group'

### ISSUE THREAD

**Issue by ChristianEdwardPadilla** [flutter/flutter#117011]
## Steps to Reproduce

1. Execute `flutter run -d chrome` on the code sample.
2. On web, use a screen reader to read the element generated by a `Semantics` widget with `link: true`.

**Expected results:**
The screen reader should indicate that the widget is a link.

**Actual results:**
The screen reader does not indicate that the widget is a link, but a group instead.

<summary>Code sample</summary>

```dart

import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());

  WidgetsFlutterBinding.ensureInitialized().setSemanticsEnabled(true);
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: Column(
        children: [
          RichText(
            text: TextSpan(
              children: [
                WidgetSpan(
                  child: Semantics(
                    button: true,
                    container: true,
                    child: const Text('role: button (works)'),
                  ),
                ),
                const TextSpan(text: '\n'),
                WidgetSpan(
                  child: Semantics(
                    link: true,
                    container: true,
                    child: const Text('role: link (does not work)'),
                  ),
                ),
              ],
            ),
          ),
        ],
      ),
    );
  }
}
```

Internal issue: b/261919978

**Comment by ChristianEdwardPadilla** [comment]
Summary of what's observed in the minimal code example:

```dart
WidgetSpan(
  child: Semantics(
    button: true,
    container: true,
    child: const Text('role: button (works)'),
    ),
),
```
Gives:
```html
<flt-semantics 
id="flt-semantic-node-5" 
aria-label="role: button (works)" 
role="button" 
tabindex="0" 
style="position: absolute; width: 624px; height: 84px; pointer-events: all; z-index: 1;">
</flt-semantics>
```


```dart
WidgetSpan(
  child: Semantics(
    link: true,
    container: true,
    child: const Text('role: link (does not work)'),
    ),
),
```
Gives:
```html
<flt-semantics
id="flt-semantic-node-6" 
aria-label="role: link (does not work)" 
role="text" 
style="position: absolute; width: 788px; height: 84px; transform-origin: 0px 0px 0px; transform: matrix(1, 0, 0, 1, 0, 84); pointer-events: all; z-index: 2;">
</flt-semantics>
```

Associated google bug: b/261919978

**Comment by huycozy** [comment]
@ChristianEdwardPadilla Thanks for filing the issue. This issue is reproducible on the latest `stable` and `master` channels with provided sample code.

<details>
<summary>flutter doctor -v (stable and master)</summary>

```bash
[✓] Flutter (Channel stable, 3.3.9, on macOS 13.0 22A380 darwin-x64, locale en-VN)
    • Flutter version 3.3.9 on channel stable at /Users/huynq/Documents/GitHub/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision b8f7f1f986 (31 hours ago), 2022-11-23 06:43:51 +0900
    • Engine revision 8f2221fbef
    • Dart version 2.18.5
    • DevTools version 2.15.0

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/huynq/Library/Android/sdk
    • Platform android-33, build-tools 31.0.0
    • ANDROID_HOME = /Users/huynq/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.0.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14A400
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)

[✓] IntelliJ IDEA Community Edition (version 2022.2.2)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 70.0.5
    • Dart plugin version 222.4167.21

[✓] IntelliJ IDEA Community Edition (version 2022.1.1)
    • IntelliJ at /Users/huynq/Library/Application Support/JetBrains/Toolbox/apps/IDEA-C/ch-0/221.5591.52/IntelliJ IDEA CE.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] VS Code (version 1.73.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.52.0

[✓] Connected device (3 available)
    • SM T225 (mobile) • R9JT3004VRJ • android-arm64  • Android 12 (API 31)
    • macOS (desktop)  • macos       • darwin-x64     • macOS 13.0 22A380 darwin-x64
    • Chrome (web)     • chrome      • web-javascript • Google Chrome 107.0.5304.110

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!

```

```bash
[!] Flutter (Channel master, 3.7.0-7.0.pre.18, on macOS 13.0.1 22A400 darwin-x64, locale en-VN)
    • Flutter version 3.7.0-7.0.pre.18 on channel master at /Users/huynq/Documents/GitHub/flutter_master
    ! Warning: `flutter` on your path resolves to /Users/huynq/Documents/GitHub/flutter/bin/flutter, which is not inside your current Flutter SDK checkout at /Users/huynq/Documents/GitHub/flutter_master. Consider adding /Users/huynq/Documents/GitHub/flutter_master/bin to the front of your path.
    ! Warning: `dart` on your path resolves to /Users/huynq/Documents/GitHub/flutter/bin/dart, which is not inside your current Flutter SDK checkout at /Users/huynq/Documents/GitHub/flutter_master. Consider adding /Users/huynq/Documents/GitHub/flutter_master/bin to the front of your path.
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision f07db4018a (12 minutes ago), 2022-12-14 03:25:49 +0200
    • Engine revision 14194c40ec
    • Dart version 3.0.0 (build 3.0.0-19.0.dev)
    • DevTools version 2.20.0
    • If those were intentional, you can disregard the above warnings; however it is recommended to use "git" directly to perform update checks and upgrades.

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/huynq/Library/Android/sdk
    • Platform android-33, build-tools 31.0.0
    • ANDROID_HOME = /Users/huynq/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14B47b
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)

[✓] IntelliJ IDEA Community Edition (version 2022.1.1)
    • IntelliJ at /Users/huynq/Library/Application Support/JetBrains/Toolbox/apps/IDEA-C/ch-0/221.5591.52/IntelliJ IDEA CE.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] VS Code (version 1.73.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.54.0

[✓] Connected device (3 available)
    • SM T225 (mobile) • R9JT3004VRJ • android-arm64  • Android 12 (API 31)
    • macOS (desktop)  • macos       • darwin-x64     • macOS 13.0.1 22A400 darwin-x64
    • Chrome (web)     • chrome      • web-javascript • Google Chrome 108.0.5359.98

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 1 category.

```
</details>

**Comment by yjbanov** [comment]
We should also check if the `Link` widget behaves correct w.r.t. to the link role.

**Comment by peixinli** [comment]
Updating bug priority to match the customer bug from google.

**Comment by math1man** [comment]
I believe this issue can be closed, as the fix in https://github.com/flutter/flutter/issues/134795 should also apply here.
